chromium_src/chrome/browser/ui/views/first_run_dialog.cc

```diff
@@ -3,18 +3,127 @@
+#include "chrome/browser/ui/browser_dialogs.h"
+#include "chrome/browser/ui/ui_features.h"
+#include "chrome/browser/ui/views/chrome_layout_provider.h"
+#include "chrome/common/url_constants.h"
 #include "chrome/grit/chromium_strings.h"
 #include "chrome/grit/generated_resources.h"
+#include "components/crash/core/app/breakpad_linux.h"
+#include "components/crash/core/app/crashpad.h"
+#include "components/strings/grit/components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/views/border.h"
+#include "ui/views/controls/button/checkbox.h"
+#include "ui/views/controls/link.h"
+#include "ui/views/layout/box_layout.h"
+#include "ui/views/layout/grid_layout.h"
+#include "ui/views/widget/widget.h"
+#include "ui/views/window/dialog_delegate.h"
+
+namespace first_run {
+
+void ShowFirstRunDialog(Profile* profile) {
+#if defined(OS_MAC)
+  if (base::FeatureList::IsEnabled(features::kViewsFirstRunDialog))
+    ShowFirstRunDialogViews(profile);
+  else
+    ShowFirstRunDialogCocoa(profile);
+#else
+  ShowFirstRunDialogViews(profile);
+#endif
+}
+
+void ShowFirstRunDialogViews(Profile* profile) {
+  FirstRunDialog::Show(profile);
+}
+
+}  // namespace first_run
+
+// static
+void FirstRunDialog::Show(Profile* profile) {
+  FirstRunDialog* dialog = new FirstRunDialog(profile);
+  views::DialogDelegate::CreateDialogWidget(dialog, NULL, NULL)->Show();
+
+  base::RunLoop run_loop(base::RunLoop::Type::kNestableTasksAllowed);
+  dialog->quit_runloop_ = run_loop.QuitClosure();
+  run_loop.Run();
+}
+
+FirstRunDialog::FirstRunDialog(Profile* profile) {
+  ALLOW_UNUSED_LOCAL(report_crashes_);
+
+  SetTitle(l10n_util::GetStringUTF16(IDS_FIRST_RUN_DIALOG_WINDOW_TITLE));
+  SetButtons(ui::DIALOG_BUTTON_OK);
+  SetExtraView(
+      std::make_unique<views::Link>(l10n_util::GetStringUTF16(IDS_LEARN_MORE)))
+      ->SetCallback(base::BindRepeating(&platform_util::OpenExternal,
+                                        base::Unretained(profile),
+                                        GURL(chrome::kLearnMoreReportingURL)));
+
+  constexpr int kChildSpacing = 16;
+  constexpr int kPadding = 24;
+
+  SetLayoutManager(std::make_unique<views::BoxLayout>(
+      views::BoxLayout::Orientation::kVertical,
+      gfx::Insets(kPadding, kPadding, kPadding, kPadding), kChildSpacing));
+
+  constexpr int kFontSize = 15;
+  int size_diff = kFontSize - views::Label::GetDefaultFontList().GetFontSize();
+  views::Label::CustomFont contents_font = {
+      views::Label::GetDefaultFontList()
+          .DeriveWithSizeDelta(size_diff)
+          .DeriveWithWeight(gfx::Font::Weight::NORMAL)};
+  auto* contents_label = AddChildView(std::make_unique<views::Label>(
+      l10n_util::GetStringUTF16(
+          IDS_FIRSTRUN_DLG_COMPLETE_INSTALLATION_LABEL_BRAVE),
+      contents_font));
+  contents_label->SetHorizontalAlignment(gfx::ALIGN_LEFT);
+  contents_label->SetMultiLine(true);
+  contents_label->SetMaximumWidth(450);
```

Comment:
nit: make this width a `constexpr` with the others

Reply:
fixed